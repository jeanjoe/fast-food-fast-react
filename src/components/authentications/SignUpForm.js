import React, { Fragment } from "react";
import {
  Container,
  Form,
  Button,
  Checkbox,
  Segment,
  Grid,
  Header,
  Message,
  Icon
} from "semantic-ui-react";

const SignUpForm = ({
  handleSubmit,
  handleOnChange,
  passwordError,
  allError,
  registerSuccess,
  loading
}) => {
  return (
    <Container>
      <Grid centered stackable>
        <Grid.Column width={6}>
          <Segment style={{ margin: "5em 0em", padding: "2em 1em" }}>
            <Form onSubmit={handleSubmit}>
              <Header textAlign="center">Create your account.</Header>
              {passwordError || allError ? (
                <Message negative>
                  <Message.Header as="h5">
                    {passwordError || allError}
                  </Message.Header>
                </Message>
              ) : (
                ""
              )}
              {registerSuccess ? (
                <Message positive>
                  <Message.Header as="h5">{registerSuccess}</Message.Header>
                </Message>
              ) : (
                ""
              )}
              <Form.Field>
                <label>First Name</label>
                <input
                  name="firstName"
                  placeholder="First Name"
                  onChange={handleOnChange}
                />
              </Form.Field>
              <Form.Field>
                <label>Last Name</label>
                <input
                  name="lastName"
                  placeholder="Last Name"
                  onChange={handleOnChange}
                />
              </Form.Field>
              <Form.Field>
                <label>Email address</label>
                <input
                  name="email"
                  placeholder="manzede@andela.com"
                  type="email"
                  onChange={handleOnChange}
                />
              </Form.Field>
              <Form.Field>
                <label>Password</label>
                <input
                  name="password"
                  placeholder="********"
                  min={8}
                  type="password"
                  onChange={handleOnChange}
                />
              </Form.Field>
              <Form.Field>
                <label>Confirm password</label>
                <input
                  name="confirmPassword"
                  placeholder="*********"
                  min={8}
                  type="password"
                  onChange={handleOnChange}
                />
              </Form.Field>
              <Form.Field>
                <Checkbox checked disabled label="I agree to the Terms and Conditions" />
              </Form.Field>
              <Button type="submit" color="red" size="medium" fluid>
                {loading ? (
                  <Fragment>
                    <Icon loading name="spinner" />
                    Signing Up...
                  </Fragment>
                ) : (
                  "Sign up"
                )}
              </Button>
            </Form>
          </Segment>
        </Grid.Column>
      </Grid>
    </Container>
  );
};

export default SignUpForm;
