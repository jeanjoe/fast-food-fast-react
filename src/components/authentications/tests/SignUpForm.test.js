import React from "react";
import expect from "expect";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import SignUpForm from "../SignUpForm";
configure({ adapter: new Adapter() });

describe("Test article component", () => {
  const props = {
    handleSubmit: jest.fn(),
    handleOnChange: jest.fn(),
    passwordError: 'error',
    loading: true,
    allError: null,
    registerSuccess: null,
  };
  const wrapper = shallow(<SignUpForm {...props} />);

  it("Matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
