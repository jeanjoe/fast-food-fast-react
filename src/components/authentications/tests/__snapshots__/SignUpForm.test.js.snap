// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test article component Matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUpForm
    allError={null}
    handleOnChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    loading={true}
    passwordError="error"
    registerSuccess={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "as": "form",
      "children": Array [
        <Header
          textAlign="center"
        >
          Create your account.
        </Header>,
        <Alert
          message="error"
          type="negative"
        />,
        "",
        <FormField
          control={[Function]}
          label="First Name"
          name="firstName"
          onChange={[MockFunction]}
          placeholder="First Name"
          type="text"
        />,
        <FormField
          control={[Function]}
          label="Last Name"
          name="lastName"
          onChange={[MockFunction]}
          placeholder="Last Name"
          type="text"
        />,
        <FormField
          control={[Function]}
          label="Email"
          name="email"
          onChange={[MockFunction]}
          placeholder="Email"
          type="email"
        />,
        <FormField
          control={[Function]}
          label="Password"
          name="password"
          onChange={[MockFunction]}
          placeholder="********"
          type="password"
        />,
        <FormField
          control={[Function]}
          label="Confirm password"
          min={8}
          name="confirmPassword"
          onChange={[MockFunction]}
          placeholder="*******"
          type="password"
        />,
        <FormField
          checked={true}
          control={[Function]}
          disabled={true}
          label="I agree to the Terms and Conditions"
        />,
        <Button
          as="button"
          color="red"
          fluid={true}
          size="medium"
          type="submit"
        >
          <React.Fragment>
            <Icon
              as="i"
              loading={true}
              name="spinner"
            />
            Signing Up...
          </React.Fragment>
        </Button>,
      ],
      "onSubmit": [MockFunction],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Create your account.",
          "textAlign": "center",
        },
        "ref": null,
        "rendered": "Create your account.",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "message": "error",
          "type": "negative",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "control": [Function],
          "label": "First Name",
          "name": "firstName",
          "onChange": [MockFunction],
          "placeholder": "First Name",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "control": [Function],
          "label": "Last Name",
          "name": "lastName",
          "onChange": [MockFunction],
          "placeholder": "Last Name",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "control": [Function],
          "label": "Email",
          "name": "email",
          "onChange": [MockFunction],
          "placeholder": "Email",
          "type": "email",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "control": [Function],
          "label": "Password",
          "name": "password",
          "onChange": [MockFunction],
          "placeholder": "********",
          "type": "password",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "control": [Function],
          "label": "Confirm password",
          "min": 8,
          "name": "confirmPassword",
          "onChange": [MockFunction],
          "placeholder": "*******",
          "type": "password",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": true,
          "control": [Function],
          "disabled": true,
          "label": "I agree to the Terms and Conditions",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "as": "button",
          "children": <React.Fragment>
            <Icon
              as="i"
              loading={true}
              name="spinner"
            />
            Signing Up...
          </React.Fragment>,
          "color": "red",
          "fluid": true,
          "size": "medium",
          "type": "submit",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Icon
                as="i"
                loading={true}
                name="spinner"
              />,
              "Signing Up...",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "loading": true,
                "name": "spinner",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "Signing Up...",
          ],
          "type": Symbol(react.fragment),
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "as": "form",
        "children": Array [
          <Header
            textAlign="center"
          >
            Create your account.
          </Header>,
          <Alert
            message="error"
            type="negative"
          />,
          "",
          <FormField
            control={[Function]}
            label="First Name"
            name="firstName"
            onChange={[MockFunction]}
            placeholder="First Name"
            type="text"
          />,
          <FormField
            control={[Function]}
            label="Last Name"
            name="lastName"
            onChange={[MockFunction]}
            placeholder="Last Name"
            type="text"
          />,
          <FormField
            control={[Function]}
            label="Email"
            name="email"
            onChange={[MockFunction]}
            placeholder="Email"
            type="email"
          />,
          <FormField
            control={[Function]}
            label="Password"
            name="password"
            onChange={[MockFunction]}
            placeholder="********"
            type="password"
          />,
          <FormField
            control={[Function]}
            label="Confirm password"
            min={8}
            name="confirmPassword"
            onChange={[MockFunction]}
            placeholder="*******"
            type="password"
          />,
          <FormField
            checked={true}
            control={[Function]}
            disabled={true}
            label="I agree to the Terms and Conditions"
          />,
          <Button
            as="button"
            color="red"
            fluid={true}
            size="medium"
            type="submit"
          >
            <React.Fragment>
              <Icon
                as="i"
                loading={true}
                name="spinner"
              />
              Signing Up...
            </React.Fragment>
          </Button>,
        ],
        "onSubmit": [MockFunction],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Create your account.",
            "textAlign": "center",
          },
          "ref": null,
          "rendered": "Create your account.",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": "error",
            "type": "negative",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "control": [Function],
            "label": "First Name",
            "name": "firstName",
            "onChange": [MockFunction],
            "placeholder": "First Name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "control": [Function],
            "label": "Last Name",
            "name": "lastName",
            "onChange": [MockFunction],
            "placeholder": "Last Name",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "control": [Function],
            "label": "Email",
            "name": "email",
            "onChange": [MockFunction],
            "placeholder": "Email",
            "type": "email",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "control": [Function],
            "label": "Password",
            "name": "password",
            "onChange": [MockFunction],
            "placeholder": "********",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "control": [Function],
            "label": "Confirm password",
            "min": 8,
            "name": "confirmPassword",
            "onChange": [MockFunction],
            "placeholder": "*******",
            "type": "password",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": true,
            "control": [Function],
            "disabled": true,
            "label": "I agree to the Terms and Conditions",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "as": "button",
            "children": <React.Fragment>
              <Icon
                as="i"
                loading={true}
                name="spinner"
              />
              Signing Up...
            </React.Fragment>,
            "color": "red",
            "fluid": true,
            "size": "medium",
            "type": "submit",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Icon
                  as="i"
                  loading={true}
                  name="spinner"
                />,
                "Signing Up...",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "loading": true,
                  "name": "spinner",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "Signing Up...",
            ],
            "type": Symbol(react.fragment),
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
